Update CSS to allow for larger screens.

syntax-error752 opened the pull request:
Allows for the scaling of the centre column in the vault.
**Before:**
<img width="1517" alt="before" src="https://user-images.githubusercontent.com/62275196/82891135-9872f880-9f90-11ea-8fed-81462470ff6f.PNG">
**After:**
<img width="1515" alt="after" src="https://user-images.githubusercontent.com/62275196/82891138-99a42580-9f90-11ea-92db-cd3a96feba67.PNG">

cscharf commented on src/scss/styles.scss:
```diff
@@ -832,3 +833,16 @@ img.logo {
 }
+
+.col-3 {
+    max-width: 240px;
```
Bootstrap uses flex-box and `%` for controlling `col-*` widths in a fluid layout. I wouldn't want to cap those simply because of the bootstrap styles that are being used for the web vault's layout without more specificity. If there are regions/parts of the web vault's layout that need to be capped on width that are currently only styles as `col-3`, for example the **FILTERS** sidebar, that should then get an additional css class appended to the markup itself, to be targeted by the stylesheet specifically:
```less
.left-nav {
    max-width: 240px;
}
```
OR, the use of specificity, nesting and *nth* operators be used for targeting those elements, i.e.
```less
.container {
    /*...*/
    .row {
        .col-3 {
            &:first-child {
                max-width: 240px;
            }
        }
    }
}
```

syntax-error752 replied:
Changes made as requested.

cscharf commented on src/scss/styles.scss:
```diff
@@ -832,3 +833,16 @@ img.logo {
+}
+
+.col-5 {
+    max-width: 420px;
```
Same

cscharf commented on src/scss/styles.scss:
```diff
@@ -832,3 +833,16 @@ img.logo {
+}
+
+.col-6 {
+    flex: 0 0 calc(100% - 500px);
```
Same here, essentially ensuring specificity to the primary content region of the web vault layout instead of any `col-6` class'd element.

syntax-error752 commented:
> I'm not sure I fully see the value in stretching the site full-width (well 90% anyway) while keeping the filters and organizations sections fluid as well (vs. fixed), since a larger screen would better allow longer organization names, other languages, folder names/structure, etc. to prevent wrapping in larger view-ports, etc. Also, `980px` is a pretty decent size as-is, I'm not sure why a fully fluid layout would be necessary for the web vault since most if not all use-cases would fail to make use of the expanded screen real-estate (can you provide a screen shot or example of where this would be more appropriate or a better UX?)

In our small business we have recently started using Bitwarden, and have imported many on the entries into our self-hosted vault.
A few people here, including myself find it a bit annoying that the vault view is limited to a small column, while all the (poorly named) entries are caused to wrap, and the space to the side gets wasted.
Ideally we would rename these, but here is a small selection of some of our long names (imported from the old password manager), copied into an otherwise empty vault (for security reasons):
<img width="583" alt="usage_1" src="https://user-images.githubusercontent.com/62275196/82982649-f73c7e80-a031-11ea-8118-633290241b1b.PNG">
Compared to expanding the columns:
<img width="848" alt="usage_2" src="https://user-images.githubusercontent.com/62275196/82982859-687c3180-a032-11ea-8afb-76f1f9b80b19.PNG">

I have implemented this is our self-hosted environment, and in our case at least, people seem to like the changes, if only because it makes the long (and poorly) named entries easier to read.

cscharf commented on src/scss/styles.scss:
```diff
@@ -188,12 +188,25 @@ input, select, textarea {
 
 .container {
-    width: 980px;
+    min-width: 980px;
```
Please move these 2 lines inside another class wrapper, `.full-width`, e.g.:
```less
.container {
    width: 980px;
    &.full-width {
        min-width: 980px;
        width: 90%;
    }
}
```

cscharf commented on src/scss/styles.scss:
```diff
@@ -188,12 +188,25 @@ input, select, textarea {
     max-width: none !important;
     margin: 0 auto;
     padding: 0;
+    .row {
```
I would think this should be removed... leaving the `.col-*` classes as-is should still allow full-width display to work w/o wrapping and we would want scaling to work across the board (not just for the center region).

cscharf commented on src/scss/styles.scss:
```diff
@@ -188,12 +188,25 @@ input, select, textarea {
+    }
 }
-
+    
```
Please remove added whitespace

cscharf commented:
@syntax-error752 , please provide localized strings only in English as part of the primary English language file, we use Crowdin upon commit/merge to `master` to source translations for those strings once published.

syntax-error752 commented:
I have converted this to an option:
![image](https://user-images.githubusercontent.com/62275196/83263399-e3477700-a201-11ea-96b3-1c89eff3daf1.png)

The scaling does not affect the top nav bar or footer.

cscharf commented:
Awesome! 2 small requests:
* Please ensure the navigation bar also scales (kind of looks funny being in the center-ish of the page otherwise)
* Please label the option: "Full Width Layout", the description underneath is perfect.

syntax-error752 commented on src/app/app.component.ts:
```diff
@@ -166,6 +172,7 @@ export class AppComponent implements OnDestroy, OnInit {
                 }
             }
         });
+        this.messagingService.send('setFullWidth');
```
Added to trigger on load.

syntax-error752 commented on src/services/htmlStorage.service.ts:
```diff
@@ -4,9 +4,9 @@ import { ConstantsService } from 'jslib/services';
 
 export class HtmlStorageService implements StorageService {
     private localStorageKeys = new Set(['appId', 'anonymousAppId', 'rememberedEmail', 'passwordGenerationOptions',
-        ConstantsService.disableFaviconKey, 'rememberEmail', 'enableGravatars', ConstantsService.localeKey,
```
Had to rearrange  to keep within the line character limit.

syntax-error752 commented on src/scss/styles.scss:
```diff
@@ -123,6 +123,18 @@ body {
+            .row {
+                .col-5 {
+                    max-width: 416px;
+                }
```
col-5 max-width set to keep the login box the same size regardless of weather on not the fill width setting is turned on.

cscharf replied:
Let's remove the override for the login form itself, if we need to reset it I would prefer it get it's own "functional class" for css targeting (e.g. "`.login-form`") but looking at it, while it's certainly a lot wider in this scenario, I believe it's acceptable without limiting its size.

syntax-error752 commented:
The requested changes have been made.

kspearrin commented on src/scss/styles.scss:
```diff
@@ -123,6 +123,18 @@ body {
         background-color: #ecf0f5;
     }
+
+    &.full-width {
```
Just add `:not(&.layout_frontend)` or something to that affect to exclude the frontend layout.

syntax-error752 commented:
Updated to use the not selector (with a pre-existing class).
